jekyll/_cci2/github-integration.adoc

```diff
@@ -0,0 +1,279 @@
+
+Next you will need to set up the necessary permissions to run your projects on CircleCI.
+
+== User keys and deploy keys
+
+**What is a user key?**
+
+A user key is user-specific an SSH key-pair. Github stores the public key, and CircleCI stores the private key. Possession of the private key gives the ability to act as that user, for purposes of 'git' access to projects.
+
+**What is a deploy key?**
+
+When you add a new project, CircleCI creates a deployment key on GitHub for your project. A deploy key is an SSH key-pair, one public, one private. GitHub stores the public key, and CircleCI stores the private key. The deployment key gives CircleCI access to a single repository. To prevent CircleCI from pushing to your repository, this deployment key is read-only.
+
+If you want to push to the repository from your builds, you will need a deployment key with write access. See the below section for GitHub-specific instructions to create a deployment key.
+
+=== Create a GitHub deploy key
+
+In this example, the GitHub repository is `https://github.com/you/test-repo`, and the CircleCI project is `https://circleci.com/gh/you/test-repo`.
+
+1. Create an SSH key-pair by following the https://help.github.com/articles/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent/[GitHub instructions]. When prompted to enter a passphrase, do **not** enter one:
+
+```shell
+ssh-keygen -t ed25519 -C "your_email@example.com"
+```
+
+1. Go to `https://github.com/you/test-repo/settings/keys`, and click **Add Deploy Key**. Enter a title in the "Title" field, then copy and paste the public key you created in step 1. Check **Allow write access**, then click **Add key**.
+
+1. Go to your project settings in the CircleCI app, select **SSH Keys**, and **Add SSH key**. In the "Hostname" field, enter `github.com`and add the private key you created in step 1. Then click **Add SSH Key**.
+
+1. In your `.circleci/config.yml` file, add the fingerprint using the `add_ssh_keys` key:
+
+```yaml
+version: 2
+jobs:
+  deploy-job:
+    steps:
+      - add_ssh_keys:
+          fingerprints:
+            - "SO:ME:FIN:G:ER:PR:IN:T"
+```
+
+When you push to your GitHub repository from a job, CircleCI will use the SSH key you added.
+
+=== How are these keys used?
+
+When CircleCI builds your project, the private key is installed into the `.ssh` directory and SSH is subsequently configured to communicate with your version control provider. Therefore, the private key is used for:
+
+- Checking out the main project
+- Checking out any GitHub-hosted submodules
+- Checking out any GitHub-hosted private dependencies
+- Automatic git merging/tagging/etc
+
+For this reason, a deploy key isn't sufficiently powerful for projects with additional private dependencies.
```

Comment:
What do they need to do then if they have private dependencies? I feel like we can signpost this better 🤔

Reply:
I think it's `docs/2.0/github-integration/#enable-your-project-to-check-out-additional-private-repositories` that they need if this is their use case